### DISCUSSION TITLE
How to run scenarios in the one yml file with the same amount of VUs?

### DISCUSSION THREAD

**Discussion by iuriigrynykha** [artilleryio/artillery#1895] [Artillery]
I have a use case when I have several scenarios under one yml file, and I need to run them all with the same amount of VUs(split equally between scenarios).
Example:
I ran a test with duration 2 and arrival rate 3, and I see the following:

**Actual**: The first scenario uses 2 users, and the second uses 4.
**Expected**: they both use 3 users.

I tried experimenting with **weight** and **rampTo** options but couldn't reach the desired effect. 
Is there any possible solution to my scenario except for splitting those scenarios between different yml files?

I really appreciate any help you can provide.

**Comment by hassy** [comment]
hi @iuriigrynykha - if you need precision then you have to split scenarios into different files. `weights` is probabilistic so setting equal weights does not guarantee equal split.

**Reply by iuriigrynykha** [reply]
Thanks for quick response!

In this case, we will need to run them all. I found a way how to do this https://github.com/artilleryio/artillery/issues/1582#issuecomment-1500066004 but I encountered issue with report. 

Is there a way to merge multiple json files into one json and build report based on it?
